src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ModelExplorer.cs

```diff
@@ -198,24 +197,38 @@ public Type ModelType
         /// Includes a <see cref="ModelExplorer"/> for each property of the <see cref="ModelMetadata"/>
         /// for <see cref="ModelType"/>.
         /// </remarks>
-        public IEnumerable<ModelExplorer> Properties
+        public IEnumerable<ModelExplorer> Properties => PropertiesInternal;
+
+        private IReadOnlyList<ModelExplorer> PropertiesInternal
```

Comment:
Could expose a new `public IReadOnlyList<ModelExplorer> PropertiesList { get; }` but `Properties` isn't referenced much. Mainly used in default display / editor templates and `ModelExplorerExtensions.GetSimpleDisplayText()`, neither of which is likely to be used more than once per request.